src/main/java/net/glowstone/io/entity/EntityStore.java

```diff
@@ -98,6 +102,37 @@ public void load(T entity, CompoundTag tag) {
             UUID uuid = UUID.fromString(tag.getString("UUID"));
             entity.setUniqueId(uuid);
         }
+
+        if (tag.isList("Passengers", TagType.COMPOUND)) {
+            for (CompoundTag entityTag : tag.getCompoundList("Passengers")) {
+                Entity passenger = loadPassenger(entity, entityTag);
+                if (passenger != null) {
+                    entity.addPassenger(passenger);
```

Comment:
I have to agree with you, that normally you would not want to fire these events during world load.  
Spigot however does throw them during world load..